### ISSUE TITLE
node_namespace_pod_container metric not available

### ISSUE THREAD

**Issue by araigneeman** [coreos/kube-prometheus#284]
Hi guys,

I just wanted to know from recents changes in grafana dashboards from this:

**namespace_pod_container:container_cpu_usage_seconds_total:sum_rate**

to :

**node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate**

The old metric is working properly on my side but not the new one, so i just wanted to know where this metric is coming from please ?

For the information i'm using the helm chart where grafana dashboards are synced from your manifests.

So for the version :
- Prometheus operator: 0.32
- Node exporter: 0.18.0
- Kube state metrics: 1.8.0
- Grafana: 6.4.2

Thanks a lot for your help

**Comment by araigneeman** [comment]
Oh yes that's the point !

I synced manually the rules with the provided script and it works now.

Thanks a lot !

**Comment by brancz** [comment]
Ok, in that case closing here.

Let us know if there is anything else! :slightly_smiling_face:

**Comment by vishnuhd** [comment]
Hi @brancz , I am still facing this issue. The query never works in prom itself :

```
sum by (namespace, pod, container) (
          rate(container_cpu_usage_seconds_total{job="kubelet", image!="", container!="POD"}[5m])
        ) * on (namespace, pod) group_left(node) max by(namespace, pod, node) (kube_pod_info)
```
It fails with below error :
```
Error executing query: found duplicate series for the match group {namespace="monitoring", pod="devqa-kube-state-metrics-d6b58d6f4-c5pzl"} on the right hand-side of the operation: [{namespace="monitoring", node="ip-11-71-21-111.ap-southeast-1.compute.internal", pod="devqa-kube-state-metrics-d6b58d6f4-c5pzl"}, {namespace="monitoring", node="ip-11-71-13-117.ap-southeast-1.compute.internal", pod="devqa-kube-state-metrics-d6b58d6f4-c5pzl"}];many-to-many matching not allowed: matching labels must be unique on one side
```
Versions :
- Helm chart version - 8.5.0
- Prom operator version - 0.34.0
- Prom version - 2.13.1
- Grafana version - 6.5.2
- Kube - 1.14
- kube-state-metrics - 2.4.1
- prometheus-node-exporter - 1.8.0 

Any help would be appreciated.

**Comment by brancz** [comment]
Does that issue persist? This sounds like either you have multiple kube state metrics pods and or Kubernetes is moving a pod. In any case the reason this is happening is because you have one time series saying the pod is on one node and one on another which is why the join is failing.

**Comment by vishnuhd** [comment]
@brancz : This still persists, there is only one kube-state-metrics pod running since days, so I don't think the pod is moving to various nodes (there are no such events as well in `kubectl describe` command).
I did a workaround by using the queries from certain other community dashboards.

**Comment by brancz** [comment]
Could you share the result of this PromQL query?

```
kube_pod_info{namespace="monitoring", pod="devqa-kube-state-metrics-d6b58d6f4-c5pzl"}
```

**Comment by vishnuhd** [comment]
@brancz : Its below one and similarly more rows :

```
kube_pod_info{cluster="devqa",created_by_kind="DaemonSet",created_by_name="aws-node",endpoint="http",exported_host_ip="10.90.23.203",exported_namespace="kube-system",exported_pod="aws-node-dgzr8",host_ip="10.90.23.247",instance="10.90.23.75:8080",job="kube-state-metrics",namespace="monitoring",node="ip-10-90-23-203.ap-southeast-1.compute.internal",pod="devqa-kube-state-metrics-d6b58d6f4-c5pzl",pod_ip="10.90.23.203",priority_class="system-node-critical",service="devqa-kube-state-metrics",uid="ade39c00-1116-11ea-887b-02a56aed46ae"}
```

**Comment by brancz** [comment]
For me to be able to troubleshoot this I’ll need to see all rows of the result.

**Comment by xiaozhangzhang1** [comment]
> The recording rule recently changed in the kubernetes-mixin, but it was adapted in the same commit. It sounds like the helm chart may not have been synced correctly. https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/274/files
> 
> I may be missing something though.

i meet the same, i use kube-promethues, release-0.6, k8s  1.18.14, node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate   can not work，

**Comment by paulfantom** [comment]
The `node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate` recording rule is shipped in `release-0.6` as seen in https://github.com/prometheus-operator/kube-prometheus/blob/release-0.6/manifests/prometheus-rules.yaml#L577-L583. If it doesn't work for you please check if it exists in your Prometheus instance and checks for potential errors in logs.

**Comment by SonalJain1707** [comment]
node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate is not working kubepromethus v0.10.0 grafana 8.3.3
